### ISSUE TITLE
Missing fastapi_users package

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal#467]
When trying to use the 1.1.0a3 client, I run into this error:

```
Traceback (most recent call last):
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/bin/fractal", line 5, in <module>
    from fractal.__main__ import run
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/__main__.py", line 3, in <module>
    from fractal.client import main
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/client.py", line 25, in <module>
    from . import cmd
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/cmd/__init__.py", line 8, in <module>
    from ._dataset import dataset_add_resource
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/cmd/_dataset.py", line 8, in <module>
    from ..common.schemas import DatasetRead
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/common/schemas/__init__.py", line 6, in <module>
    from .user import *  # noqa: F403
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/common/schemas/user.py", line 4, in <module>
    from fastapi_users import schemas
ModuleNotFoundError: No module named 'fastapi_users'
Traceback (most recent call last):
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/bin/fractal", line 5, in <module>
    from fractal.__main__ import run
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/__main__.py", line 3, in <module>
    from fractal.client import main
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/client.py", line 25, in <module>
    from . import cmd
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/cmd/__init__.py", line 8, in <module>
    from ._dataset import dataset_add_resource
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/cmd/_dataset.py", line 8, in <module>
    from ..common.schemas import DatasetRead
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/common/schemas/__init__.py", line 6, in <module>
    from .user import *  # noqa: F403
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/common/schemas/user.py", line 4, in <module>
    from fastapi_users import schemas
ModuleNotFoundError: No module named 'fastapi_users'
Traceback (most recent call last):
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/bin/fractal", line 5, in <module>
    from fractal.__main__ import run
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/__main__.py", line 3, in <module>
    from fractal.client import main
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/client.py", line 25, in <module>
    from . import cmd
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/cmd/__init__.py", line 8, in <module>
    from ._dataset import dataset_add_resource
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/cmd/_dataset.py", line 8, in <module>
    from ..common.schemas import DatasetRead
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/common/schemas/__init__.py", line 6, in <module>
    from .user import *  # noqa: F403
  File "/Users/joel/opt/miniconda3/envs/fractal-client-1.1.0a3/lib/python3.9/site-packages/fractal/common/schemas/user.py", line 4, in <module>
    from fastapi_users import schemas
ModuleNotFoundError: No module named 'fastapi_users'
```

Do we need to add this to the dependencies or drop it?

**Comment by tcompa** [comment]
Consequence of https://github.com/fractal-analytics-platform/fractal/pull/455. I'll check it out

**Comment by tcompa** [comment]
This was my bad - I'm now re-adding the fastapi_users dependency.

### Explanation

This grep output
```
tommaso@exact:~/Fractal/fractal$ pwd
/home/tommaso/Fractal/fractal
tommaso@exact:~/Fractal/fractal$ git grep fastapi_users
poetry.lock:    {file = "fastapi_users-10.4.0-py3-none-any.whl", hash = "sha256:c8edc0c16b3bca41c4b4cea479b85b8f37f6682414e8649e01a05c0b135dbe7d"},
poetry.lock:    {file = "fastapi_users-10.4.0.tar.gz", hash = "sha256:7f77a2a74be8833daaba5d60f8e8c5b445bc26f26aecc445eea9d80e877d5629"},
poetry.lock:    {file = "fastapi_users_db_sqlmodel-0.2.0-py3-none-any.whl", hash = "sha256:b4e91fe9afe719040888ad886ea033b8e1f8fcd459b242733e95e33cfa16984c"},
```
suggests that we are never importing this library.
However (here's the beauty of git submodules...), `git grep` does not enter in `common`. Doing it explicitly shows that we do need that package.
```
tommaso@exact:~/Fractal/fractal$ cd fractal/common/
tommaso@exact:~/Fractal/fractal/fractal/common$ git grep fastapi_users
schemas/user.py:from fastapi_users import schemas
```

**Comment by tcompa** [comment]
As part of #468, I added a new `pip_install` github action that does the following:
* `pip install -e .`
* `fractal -h`

In case of missing dependencies (e.g. because of subtleties of git submodules), this will fail as in https://github.com/fractal-analytics-platform/fractal/actions/runs/4435354824/jobs/7782406288:
```
Run fractal -h
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.2/x64/bin/fractal", line 5, in <module>
    from fractal.__main__ import run
  File "/home/runner/work/fractal/fractal/fractal/__main__.py", line 3, in <module>
    from fractal.client import main
  File "/home/runner/work/fractal/fractal/fractal/client.py", line 25, in <module>
    from . import cmd
  File "/home/runner/work/fractal/fractal/fractal/cmd/__init__.py", line 8, in <module>
    from ._dataset import dataset_add_resource
  File "/home/runner/work/fractal/fractal/fractal/cmd/_dataset.py", line 8, in <module>
    from ..common.schemas import DatasetRead
  File "/home/runner/work/fractal/fractal/fractal/common/schemas/__init__.py", line 6, in <module>
    from .user import *  # noqa: F403
    ^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/fractal/fractal/fractal/common/schemas/user.py", line 4, in <module>
    from fastapi_users import schemas
ModuleNotFoundError: No module named 'fastapi_users'
Error: Process completed with exit code 1.
```

**Comment by tcompa** [comment]
For the record: the actual CI runs with `poetry install --with dev`, and then installs fastapi-users via fractal-server. This is why we couldn't catch the error earlier.

**Comment by tcompa** [comment]
Fixed in `1.1.0a4`.

**Comment by jluethi** [comment]
Ah, nice to know how this passed the tests and good to have this extra level of testing for pip installation then! :)
